src/editions/index.md

```diff
@@ -42,27 +42,6 @@ You only need to upgrade if you want to take advantage of such features.
 ## Trying out the 2018 edition
 
 At the time of writing, there are two editions: 2015 and 2018. 2015 is today's
-Rust; Rust 2018 will ship later this year. To give Rust 2018 a try, you can
-add this feature to your crate root:
-
-```rust
-// Opt in to unstable features expected for Rust 2018
-#![feature(rust_2018_preview)]
```

Comment:
This code snippet gave the user a crash-course into the new edition.
It is sorta useful if all the user wants to do is test it out but not migrate yet.
I think we should somehow retain the mention of `#![feature(rust_2018_preview)]` earlier than 1.1 for now.

Reply:
Unfortunately there's a few part of things I think are misleading:

* This is recommending turning on the `rust_2018_idioms` lint without properly warning about how underbaked it currently is (huge numbers of warnings and warnings with no real actionable advice)
* It's enabling the 2018 edition without going through `cargo fix` which can give people the impression that their crates' hundreds of errors (due to paths) mean the edition is a huge hazard
* It's jumping the gun a bit in terms of what this all is and how it all interacts with one another.

This is all explained in the transition guide (hopefully in a relatively "quick" fashion) though! I just wanted to make sure we're very careful in this regard as this is a very sensitive topic (migrating to the new edition) and one that I think is important to get right

Reply:
Fair enough :)